### PR TITLE
fix: sqlite bug with getListDatabasesSQL

### PR DESCRIPTION
**Pull request by iCodr8**
fixed the folloging sqlite bug
Doctrine\DBAL\Exception: Operation 'Doctrine\DBAL\Platforms\AbstractPlatform::getListDatabasesSQL' is not supported by platform.

### CODE REVIEW

**Comment by iCodr8** [comment]
It's only a temporary fix.

**Comment by iCodr8** [comment]
See: https://github.com/liip/LiipTestFixturesBundle/issues/154

File: src/Services/DatabaseTools/ORMDatabaseTool.php [hunk 1]
```diff
@@ -160,7 +161,11 @@ protected function createDatabaseIfNotExists(): void
         $tmpConnection = DriverManager::getConnection($params);
         $tmpConnection->connect();
 
-        if (!\in_array($dbName, $tmpConnection->getSchemaManager()->listDatabases(), true)) {
+        $databases = $tmpConnection->getDatabasePlatform() instanceof SqlitePlatform
+            ? ['']
```
Comment (alexisLefebvre) [inline src/Services/DatabaseTools/ORMDatabaseTool.php:165]:
```suggestion
            ? []
```

I think we don't need an empty element.

**Comment by AlexandreGerault** [comment]
This code is like the one I proposed to undestand the issue but I really think it shouldn't be merged as it is really unclean. What if another ORMDatabaseTool does not support a specific operation?

The issue comes, in my opinion, from the way Dbal has been architectured. An abstract class defining an implementation of throwing exception shouldn't exist (it means the mother class define a function, but that might not be implemented in children). Only valid classes should exists (and I think this is one of the SOLID principles, maybe Liskov's subtitution).

**Comment by alexislefebvre** [comment]
Thanks for the PR, it helped me to understand the issue.

We may test that `$tmpConnection->getSchemaManager()` has a `listDatabases` method.

**Comment by alexislefebvre** [comment]
> We may test that `$tmpConnection->getSchemaManager()` has a `listDatabases` method.

Nope that's wrong, we have to understand why `ORMSqliteDatabaseTool` is not used then a SQLite database is configured.

**Comment by alexislefebvre** [comment]
Let's investigate on a possible side effect of config: https://github.com/liip/LiipTestFixturesBundle/issues/154#issuecomment-1027371322

**Pull request closed without merging**